os-eap-launch/added/launch/jgroups.sh

```diff
@@ -21,22 +21,51 @@ function configureEnv() {
 function configure_jgroups_encryption() {
   jgroups_encrypt=""
 
-  if [ -n "${JGROUPS_ENCRYPT_SECRET}" ]; then
-    if [ -n "${JGROUPS_ENCRYPT_NAME}" -a -n "${JGROUPS_ENCRYPT_PASSWORD}" ] ; then
-      # For new JGroups we need to use SYM_ENCRYPT protocol
-      jgroups_encrypt="\
-        <protocol type=\"SYM_ENCRYPT\">\
-          <property name=\"provider\">SunJCE</property>\
-          <property name=\"sym_algorithm\">AES</property>\
-          <property name=\"encrypt_entire_message\">true</property>\
-          <property name=\"keystore_name\">${JGROUPS_ENCRYPT_KEYSTORE_DIR}/${JGROUPS_ENCRYPT_KEYSTORE}</property>\
-          <property name=\"store_password\">${JGROUPS_ENCRYPT_PASSWORD}</property>\
-          <property name=\"alias\">${JGROUPS_ENCRYPT_NAME}</property>\
-        </protocol>"
-    else
-      log_warning "Partial JGroups encryption configuration, the communication within the cluster WILL NOT be encrypted."
-    fi
-  fi
+  JGROUPS_ENCRYPT_PROTOCOL="${JGROUPS_ENCRYPT_PROTOCOL:=SYM_ENCRYPT}"
+
+  case "${JGROUPS_ENCRYPT_PROTOCOL}" in
+    "SYM_ENCRYPT")
+      log_info "Configuring JGroups cluster traffic encryption protocol to SYM_ENCRYPT."
+      local JGROUPS_UNENCRYPTED_MESSAGE="Detected <STATE> JGroups encryption configuration, the communication within the cluster WILL NOT be encrypted."
+      local KEYSTORE_ERROR_MESSAGE=""
+      [ -n "${JGROUPS_ENCRYPT_SECRET}" -a -n "${JGROUPS_ENCRYPT_NAME}" -a -n "${JGROUPS_ENCRYPT_PASSWORD}" ]
```

Comment:
@iankko Why do we have JGROUPS_ENCRYPT_SECRET env var? Does it something somewhere? In this script it's only check whether it exists and it has to exist, but no useful information is read from it. Also what if user doesn't specify JGROUPS_ENCRYPT_KEYSTORE_DIR and JGROUPS_ENCRYPT_KEYSTORE which are used for configuration?

Reply:
@Crumby JGROUPS_ENCRYPT_SECRET is a reference to the secret installed by the app, template example here: https://github.com/jboss-container-images/jboss-eap-7-openshift-image/blob/eap-cd-dev/templates/eap-cd-sso-s2i.json#L828

I think the secret is a prerequisite for having encryption in jgroups.

Reply:
@Crumby That's actually a very good question (re: why do we have JGROUPS_ENCRYPT_SECRET). I would assume this is needed in the templates to be able to mount the [`eap-jgroups-keystore-volume`](https://github.com/jboss-openshift/application-templates/blob/master/eap/eap71-https-s2i.json#L608). But you are correct, even if supplied it's not used for actual setting of JGroups protocol type.

@rcernich -^ Can you clarify why it was / is still needed? IOW is the only usage of the `JGROUPS_ENCRYPT_SECRET` variable to specify the secret, to be used as the source for the [volume mount](https://docs.openshift.com/container-platform/3.10/dev_guide/volumes.html#adding-volumes) (see `-t` option in `Table 2. Supported options for Adding Volumes` in the previous link). Would it be possible to use configmap in the same way? If so, should we enhance the support?

@Crumby For your second question (re: `Also what if user doesn't specify JGROUPS_ENCRYPT_KEYSTORE_DIR and JGROUPS_ENCRYPT_KEYSTORE which are used for configuration?`)

From looking at the template, the `JGROUPS_ENCRYPT_KEYSTORE_DIR` is set [always](https://github.com/jboss-openshift/application-templates/blob/master/eap/eap71-https-s2i.json#L574), when using the template. But `JGROUPS_ENCRYPT_KEYSTORE` might be undefined (isn't required param AFAICT). In that case the volume mount would fail, and the pod would get stuck in 'Pending' state. For deploying the image directly, both of them need to be specified (if not, again the pod would be stuck keep trying to mount the volume).



 